src/wpool/mongoose_gun_worker.erl

```diff
@@ -0,0 +1,262 @@
+%%%-------------------------------------------------------------------
+%%% @doc
+%%% This module a wpool worker capable of establishing HTTP/2
+%%% connections using the `Gun' library.
+%%% @end
+%%%-------------------------------------------------------------------
+-module(mongoose_gun_worker).
+
+-behaviour(gen_server).
+
+%% API
+-export([start_link/2]).
+
+%% gen_server callbacks
+-export([init/1,
+         handle_continue/2,
+         handle_call/3,
+         handle_cast/2,
+         handle_info/2]).
+
+-record(state, {host :: inet:hostname() | inet:ip_address(),
```

Comment:
`state` is a too common state name, there's a bazillion erlang modules defining a different `state` record, which makes tracing more confusing and doing `rr(state)` in the shell impossible, please use some more descriptive name like `gun_worker_state` or `http_worker_state` or so.